documentation/modules/ROOT/pages/transformations/outbox-event-router.adoc

```diff
@@ -359,6 +360,7 @@ The SMT automatically filters out `DELETE` operations on an outbox table.
 |`aggregateid`
 |Table
 |Specifies the outbox table column that contains the payload ID.
+This ID will be used as the emitted event's key.
```

Comment:
Is this not rather driven via `table.field.event.key`? It's the duplication between `table.field.event.key` and `table.field.event.payload.id` which I'm doubtful about. Both default to the aggregate id.

Having `table.field.event.id` for propagating the event id to a header property makes sense (and unlike what I said on chat, I think it's good to have it as a bespoke option; while it _could_ be represented via the generic property propagation mechanism, this seems to be a generally desireable default behavior).

Reply:
> It's the duplication between table.field.event.key and table.field.event.payload.id which I'm doubtful about. Both default to the aggregate id.

So comparing the documentation to the code, there shouldn't be a default value for `table.field.event.key` as it can safely be left unsupplied and the SMT will default to using `table.field.event.payload.id`.  Now as to why they two exist in the first place, I'm not entirely sure as this was there from the start.  @renatomefi since you contributed this originally, do you have any insight to share on why the two properties exist?  There must have been a technical reason.

Reply:
So the event key is basically the kafka message key, meaning it'll be used for partitioning/ordering all that stuff.
The event ID itself is unique for all events, this can be used for deduplication downstream, as you might have many events with the same key. It's in the header to be used as metadata and not as a message key, because ordering will be screwed up by it given it's just a random uuid for each event.

Reply:
Thanks, @renatomefi that makes sense explained that way.  So I think we need to convey this in the docs and likely in the field descriptions in the configuration class, would you agree @gunnarmorling ?

Reply:
Sorry, it is still not clear to me. As you say, @renatomefi, there's _two_ pieces of information which need to be propagated:

* The Kafka message key; based on the aggregate id, so to ensure correct ordering of all messages for the same aggregate
* The event id; propagated as a header, used downstream for de-duplication

But currently, there are _three_ related settings:

* table.field.event.id (defaults to `id`)
* table.field.event.key (defaults to `aggregateid`)
* table.field.event.payload.id (defaults to `aggregateid`)

I can make sense out of the first two, they map nicely to the two things we need. But what's the purpose of that third one? As per @Naros' comment, the last two seem redundant, and one could go away (with deprecation of course), unless I am missing something. I'd vote for keeping `table.field.event.key`, as its name is more expressive. @renatomefi, @Naros, any thoughts?

Reply:
Ok, so after taking another look I think we can safely deprecate `table.field.event.payload.id` and ignore it. It's solely used as a fallback if `table.field.event.key` isn't given. I don't think this provides any value. @Naros, can you do that change for the two SMTs? No need to deprecate first for the MongoDB one of course, we just remove it there. Thx!